sycl/include/sycl/ext/intel/experimental/kernel_properties.hpp

```diff
@@ -10,72 +10,58 @@
 
 #pragma once
 
-#include <sycl/ext/intel/experimental/detail/misc_intrin.hpp>
-
-#include <sycl/detail/boost/mp11.hpp>
+#include <sycl/ext/oneapi/properties/property.hpp>
+#include <sycl/ext/oneapi/properties/property_value.hpp>
 
 namespace sycl {
 __SYCL_INLINE_VER_NAMESPACE(_V1) {
-namespace ext::intel::experimental {
-
-namespace kernel_properties {
-
-/// <summary>
-///  This namespace contains APIs to set kernel properties.
-/// </summary>
+namespace ext {
+namespace intel {
+namespace experimental {
 
-// Implementation note: <property_class>::value fields should match property IDs
-// specified in llvm/lib/SYCLLowerIR/LowerKernelProps.cpp
-
-namespace detail {
-// Proxy to access private property classes' fields from the API code.
-template <class T> struct proxy {
-  static constexpr int value = T::value;
+enum class RegisterAllocMode : uint32_t {
```

Comment:
Is this API a part of some spec? If not, we should not expose this publicly (and should hide into detail:: namespace).

Reply:
see discussion above